util/configure_jep_native_binaries.py

```diff
@@ -35,91 +43,109 @@
+            return [f.locate() for f in pkg_files if f.match(pattern)]
+
+    try:
+        import pkg_resources
+    except ImportError:
+        pass
+    else:
+        dist = pkg_resources.get_distribution(pkgname)
+        if dist:
+            pkg_dir = Path(dist.location) / dist.key
+            return pathglob(pkg_dir, pattern)
 
-    we use naive method of checking each path in sys.path for a folder named jep
-    """
     for path in sys.path:
-        jep_dir = Path(path) / JEP_PY_FOLDER_NAME
-        logger.debug("Checking if %s exists" % jep_dir)
-        if jep_dir.is_dir():
-            return jep_dir
-    return Path()
+        pkg_dir = Path(path) / pkgname
+        logger.debug("Checking if %s exists" % pkg_dir)
+        if pkg_dir.is_dir():
+            return pathglob(pkg_dir, pattern)
+
+    if fallback_dir:
+        return pathglob(fallback_dir, pattern)
+    raise ValueError("could not determine paths for package %s" % pkgname)
+
 
+def copyfile(src, dest):
+    logger.info("Copying %s to %s" % (src, dest))
+    shutil.copy(str(src), str(dest), follow_symlinks=True)
 
 def main(args):
     """ """
     if args.debug:
         logger.setLevel(logging.DEBUG)
-
-    if sys.platform in ("darwin",):
-        # we haven't tested MacOS enough to know for sure if we fully support it
-        logger.error("MacOS is not supported!")
-        return -1
-
-    if sys.platform in ("win32", "cygwin"):
-        logger.debug("Detected Windows OS")
-
-        os_lib_name = JEP_OS_LIB_NAME_WINDOWS
-        os_lib_path = Path(GHIDRA_OS_LIB_PATH_WINDOWS)
-    else:
-        logger.debug("Detected Linux OS")
-
-        os_lib_name = JEP_OS_LIB_NAME_LINUX
-        os_lib_path = Path(GHIDRA_OS_LIB_PATH_LINUX)
-
-    logger.info("Searching for Jep Python module directory")
-
     if args.path:
-        jep_dir = Path(args.path)
-        if not jep_dir.is_dir():
-            logger.error("Python module directory %s does not exist!" % args.path)
-            return -1
+        fallback_dir = Path(args.path)
     else:
-        jep_dir = find_jep_dir()
-        if not jep_dir:
-            logger.error("Could not find Jep Python module directory!")
-            return -1
+        fallback_dir = None
 
-    logger.info("Found Jep Python module directory at %s" % jep_dir)
+    # Get Jep OS names
+    if sys.platform in JEP_OS_LIB_NAMES:
+        jep_os_lib_prefix, jep_os_lib_ext = JEP_OS_LIB_NAMES[sys.platform]
+    else:
+        logger.error("Unsupported platform: %s" % sys.platform)
+        return -1
 
-    try:
-        jep_java_lib_name = glob.glob(str(Path(jep_dir) / "*.jar"), recursive=False)[0]
-    except IndexError:
-        logger.error("Could not find Jep JAR file in directory %s" % jep_dir)
+    # Get Ghidra names
+    ghidra_lib_path = Path(GHIDRA_JAVA_LIB_PATH)
+    if sys.platform in GHIDRA_OS_LIB_NAMES:
+        ghidra_os_lib_path = Path(GHIDRA_OS_LIB_PATH) / (GHIDRA_OS_LIB_NAMES[sys.platform] + os.uname().machine)
```

Comment:
`os.uname` does not work on Windows - Ghidra only supports 64-bit so we can be confident using hard-coded paths for the `os` directory e.g. `win_x86_64`. We could include the hard-coded paths for each supported OS in `GHIDRA_OS_LIB_NAMES`.

Reply:
True but Ghidra only supports a subset of those architectures: https://github.com/NationalSecurityAgency/ghidra/blob/c51183f1a262c80d45aac48d4f3343c67b6bd41a/GPL/nativePlatforms.gradle#L21-L28. We just need to make sure that the Jep native binaries are copied to the appropriate `os` directory where Ghidra knows to look e.g. `win_x86_x64`.